test/cbmc/proofs/setDataInterface/setDataInterface_harness.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * AWS IoT Over-the-air Update v3.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file setDataInterface_harness.c
+ * @brief Implements the proof harness for setDataInterface function.
+ */
+/* Ota interface includes. */
+#include "ota_interface_private.h"
+
+void setDataInterface_harness()
+{
+    OtaDataInterface_t * pDataInterface;
+    OtaDataInterface_t dataInterface;
+    OtaErr_t err;
+
+    uint8_t * pProtocol;
+    size_t protocolBufferSize;
+
+    /* The maximum size of the protocol buffer is defined by OTA_PROTOCOL_BUFFER_SIZE
+     *  and thus size variable cannot exceed OTA_PROTOCOL_BUFFER_SIZE. */
+    __CPROVER_assume( protocolBufferSize <= OTA_PROTOCOL_BUFFER_SIZE );
```

Comment:
This is not the maximum size, it's the exact size. So I don't think this makes sense to do. There may be other places where this macro is used and that could cause issues. For example, if there is a for loop that iterates up until the OTA_PROTOCOL_BUFFER_SIZE.

In this case, if the protocolBufferSize was smaller, which is possible with this code, there would be a segmentation fault as we step out of bounds of the array.

This is not a user config, so it won't change depending on the user's application and it won't change at runtime. In this case we can just allocate a static buffer of size OTA_PROTOCOL_BUFFER_SIZE. I noticed that the array is statically allocated in the ota.c source file, so I think this okay to do.

When you change this to static allocation:
1) Add a comment about how this buffer is statically allocated inside of the library source code based on macro that is not configurable by the user, so we know that it will always be this size.
2) Make sure to remove the free statement that you have.
3) Delete the unnecessary variables, namely protocolBufferSize.

Reply:
Updated the file in 6584e87[https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/245/commits/6584e87f79a86a2e5943af583d8daa258dd66ee5]